### ISSUE TITLE
Implement NAXSI as WAF 

### ISSUE THREAD

**Issue by flo-mic** [linuxserver/docker-swag#105]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Desired Behavior
It would be great to have a WAF (Web application firewall) inside this image to improve the security even more. I would suggest to use [naxsi](https://github.com/nbs-system/naxsi) as there is already a [apk package](https://pkgs.alpinelinux.org/package/edge/main/x86/nginx-mod-http-naxsi) available and it is "just" the configurational part and some predefined rules missing.

If you agree that such a feature could be a benefit for this project I would integrate this and create a PR. Just want to be sure that such a PR will be accepted if it is well thought out, before I spend my time on this 😉 

My plan would be to integrate naxsi in the main container and provide some basic rule pattern from the official community github page. If this is working I would try to create a docker mod with [nxtool](https://github.com/nbs-system/naxsi/tree/master/nxapi) so that there is also a possibility to work on naxsi rules with a web interface.

## Current Behavior
Currently no WAF which could protect the webserver from XSS and SQL Incections.

## Alternatives Considered
As alternative we could think about ModSecurity, but honestly this is a lot of work as we would need to build nginx from source files and this might cause hugh workload with upcomming releases and potential changes. With naxsi we can use an already existing apk package 😉

**Comment by flo-mic** [comment]
So just an update on this, I started implementing this as I realy want this feature to improve the security. Just hope that you will accept the PR later 😉 

So far naxsi is integrated in my dev environment with some sample rules from the official as from the community repository of naxsi. Also a file comparsion is already there to inform users about new updates if the files integrated sample rules from the source repos were updated. 

The next big step is to integrate the automatic rule creater which learns with the naxsi rules so that the users do not need to create rules on their own. Also a integration into fail2ban will come.

Of course all my modifications will be optional to avoid any breaking changes on existing installations but brings a improved security for people like me that can't make things secure enough 😄

**Comment by aptalca** [comment]
better make the whole thing a mod

**Comment by flo-mic** [comment]
Ok, I switched to a docker mod as suggested. It is anyway better because there is a lot of readme needed to explain how to configure naxsi to work properly 😉 Especially when I start explaining the rule behavior. 

For now the basic configuration is included in the docker mod and naxsi is already blocking requests. Next steps will be to have the rule generator as mentioned.

If someone is interested here is already a first version. But do not use this productive as it will block to many requests as soon as you disable the learning mode and the rule creator for learning is still missing.

I will close this issue now as this will be handled in the docker-mods repository. I will open a PR for the mod as soon as I finished the whole implementation.
